ShopGo/NavigationManager.swift

```diff
@@ -0,0 +1,54 @@
+//
+//  NavigationManager.swift
+//  ShopGo
+//
+//  Created by Quratulain on 8/5/23.
+//
+
+import UIKit
+
+enum AppScreen {
+    case LoginSignup
+    case Onboarding
+    case CreateAccount
+    case Login
+    case Terms
+    case Home
+    case MyProfile
+    case OTP
+    
+    var viewControllerIdentifier: String {
+        switch self {
+        case .LoginSignup:
+            return "LoginSignupViewController"
+        case .Onboarding:
+            return "OnBoardingPageViewController"
+        case .CreateAccount:
+            return "CreateAccountViewController"
+        case .Login:
+            return "LoginViewController"
+        case .Terms:
+            return "TermsViewController"
+        case .Home:
+            return "HomeViewController"
+        case .MyProfile:
+            return "MyprofileViewController"
+        case .OTP:
+            return "OTPViewController"
+        }
+    }
+}
+
+class NavigationManager {
+    func navigate(to screen: AppScreen, in storyboardName: String,from viewController: UIViewController) {
+        
+        let destinationViewController: UIViewController
+        
+        switch screen {
+        case .LoginSignup, .Onboarding, .CreateAccount, .Home, .Login, .MyProfile, .OTP, .Terms:
+            let storyboard = UIStoryboard(name: storyboardName, bundle: nil)
+            destinationViewController = storyboard.instantiateViewController(withIdentifier: screen.viewControllerIdentifier)
+        }
+        viewController.navigationController?.pushViewController(destinationViewController, animated: true)
```

Comment:
Move this inside curly brackets under line 50.

Reply:
@aliahmed1105 Placing this line inside the switch statement would cause a compilation error because the destinationViewController is only defined within the scope of the switch case where it's created. It would not be accessible outside of that case, and the compiler would raise an error.